File: core/internal/gamemanagement/dataaccess/game.go
package dataaccess

import "gorm.io/gorm"

// Game is the main game
type Game struct {
	gorm.Model
	//ID    uint   `json:"id" gorm:"primary_key"`
	Title       string `json:"title"`
	Description string
	Code        string `json:"author"`
	Status      string `json:"status"`
}

File: flutter/secretsanta/lib/model/game.dart
import 'package:secretsanta/model/status.dart';

class Game {
  final String code;
  final String name;
  final String description;
  final Status status;
  final String gifted;

  Game(this.code, this.name, this.description, this.status, this.gifted);

  factory Game.fromMap(Map<String, dynamic> json) {
    return Game(
      json['code'],
      json['name'],
      json['description'],
      json['status'],
      json['gifted'],
    );
  }

  Map<String, dynamic> toMap() {
    return {
      "code": this.code,
      "name": this.name,
      "description": this.description,
      "status": this.status,
      "gifted": this.gifted,
    };
  }
}
